### ISSUE TITLE
How to install pip?

### ISSUE THREAD

**Issue by wannaphong** [RustPython/RustPython#2545]
Hello. I cannot install pip to RustPython. I use https://github.com/RustPython/RustPython/commit/b870fbef5dba789e07d04442bcc00732743849a7. Can you guide me?

```
# rustpython get-pip.py
ERROR: Exception:
Traceback (most recent call last):
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/resolvers.py", line 171, in _merge_into_criterion
    crit = self.state.criteria[name]
KeyError: 'pip'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/cli/base_command.py", line 189, in _main
    status = self.run(options, args)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/cli/req_command.py", line 184, in wrapper
    raise
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/cli/req_command.py", line 178, in wrapper
    return func(self, options, args)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/commands/install.py", line 445, in run
    return ERROR
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/commands/install.py", line 317, in run
    reqs, check_supported_wheels=not options.target_dir
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/resolution/resolvelib/resolver.py", line 127, in resolve
    six.raise_from(error, e)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/resolution/resolvelib/resolver.py", line 122, in resolve
    requirements, max_rounds=try_to_avoid_resolution_too_deep,
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/resolvers.py", line 453, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/resolvers.py", line 320, in resolve
    raise ResolutionImpossible(e.criterion.information)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/resolvers.py", line 318, in resolve
    name, crit = self._merge_into_criterion(r, parent=None)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/resolvers.py", line 173, in _merge_into_criterion
    crit = Criterion.from_requirement(self._p, requirement, parent)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/resolvers.py", line 82, in from_requirement
    if not cands:
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/resolvelib/structs.py", line 124, in __bool__
    return bool(self._sequence)
  File "/root/RustPython/vm/pylib-crate/Lib/functools.py", line 593, in wrapper
    result = user_function(*args, **kwds)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/resolution/resolvelib/found_candidates.py", line 143, in __bool__
    return any(self)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/resolution/resolvelib/found_candidates.py", line 35, in _iter_built
    for version, func in infos:
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/resolution/resolvelib/factory.py", line 223, in iter_index_candidate_infos
    hashes=hashes,
  File "/root/RustPython/vm/pylib-crate/Lib/functools.py", line 569, in wrapper
    result = user_function(*args, **kwds)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/package_finder.py", line 872, in find_best_candidate
    candidates = self.find_all_candidates(project_name)
  File "/root/RustPython/vm/pylib-crate/Lib/functools.py", line 569, in wrapper
    result = user_function(*args, **kwds)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/package_finder.py", line 816, in find_all_candidates
    project_url, link_evaluator=link_evaluator,
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/package_finder.py", line 779, in process_project_url
    html_page = self._link_collector.fetch_page(project_url)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/collector.py", line 617, in fetch_page
    return _get_html_page(location, session=self.session)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/collector.py", line 452, in _get_html_page
    _handle_get_page_fail(link, "timed out")
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/collector.py", line 429, in _get_html_page
    resp = _get_html_response(url, session=session)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/index/collector.py", line 143, in _get_html_response
    "Cache-Control": "max-age=0",
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/requests/sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_internal/network/session.py", line 424, in request
    return super().request(method, url, *args, **kwargs)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/cachecontrol/adapter.py", line 53, in send
    resp = super(CacheControlAdapter, self).send(request, **kw)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/requests/adapters.py", line 531, in send
    raise
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connectionpool.py", line 776, in urlopen
    self._put_conn(conn)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connectionpool.py", line 761, in urlopen
    err = e
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connectionpool.py", line 386, in _make_request
    raise
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/connection.py", line 394, in connect
    cert_reqs=resolve_cert_reqs(self.cert_reqs),
  File "/tmp/tmprsegwb7e/pip.zip/pip/_vendor/urllib3/util/ssl_.py", line 303, in create_urllib3_context
    context.options |= options
  File "/root/RustPython/vm/pylib-crate/Lib/ssl.py", line 602, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
AttributeError: 'getset_descriptor' object has no attribute '__set__'
```

**Comment by coolreader18** [comment]
Ah, I just ran into this too, it should be fixed on the `coolreader18/misc-fixes` branch/PR #2544

**Comment by coolreader18** [comment]
Fixed, and also rustpython now has a `--install-pip` option to make it super easy to install pip.
